process/block/baseProcess.go

```diff
@@ -988,6 +988,16 @@ func (bp *baseProcessor) saveBody(body *block.Body) {
 		log.Trace("saveBody.Put -> MiniBlockUnit", "time", time.Since(startTime))
 	}
 
+	finalReceiptHash, marshalizedReceiptsHashes, errNotCritical := bp.txCoordinator.CreateMarshalizedReceipts()
+	if errNotCritical != nil {
+		log.Warn("saveBody.CreateMarshalizedReceipts", "error", errNotCritical.Error())
+	}
+
+	errNotCritical = bp.store.Put(dataRetriever.ReceiptsUnit, finalReceiptHash, marshalizedReceiptsHashes)
```

Comment:
the problem is not resolved. The saving key for this data structure should come from shardHdr.ReceiptsHash not a new computed hash. otherwise you will not find it,

Reply:
Actually is not a new computed, it will be exactly the one from shardHdr as this check and the hash calculation is already done in ProcessBlock by calling VerifyCreatedBlockTransactions wich uses the same pattern.